### ISSUE TITLE
[BUG]

### ISSUE THREAD

**Issue by akrabu** [aonez/Keka#1318]
### Configuration
- Keka version: 1.3.3 (5230)
- macOS version: Ventura 13.4.1 (c)

### Describe the bug
I was creating about 20 7z AES encrypted archives, with splitting turned on, and verification.  One did not verify properly.  When tested using 7z-cli, it checked out just fine. 

### To Reproduce
 This happens sporadically, and I'm not certain I can reproduce it.

### Expected behavior
Verification to succeed, provided it would succeed using 7z-cli.

### Screenshots

![Screenshot 2023-08-26 at 12 38 58 PM](https://github.com/aonez/Keka/assets/6868732/6f471903-60df-4514-a545-4125850efe8b)

![Screenshot 2023-08-26 at 12 39 31 PM](https://github.com/aonez/Keka/assets/6868732/8c335bc4-5ea2-4747-859c-d1808ae228bd)

[keka.log](https://github.com/aonez/Keka/files/12446626/keka.log)

**Comment by aonez** [comment]
Did you used a password for all the operations? Was it the same password on all of them?

**Comment by akrabu** [comment]
Sure did, and yes it was.  I put in a password, and drug them all in at once.

Ps. Thanks for the follow-back on Mastadon haha :)

**Comment by aonez** [comment]
Just one last question to try to replicate, were all the operations ~40GB?

**Comment by akrabu** [comment]
<img width="316" alt="Screenshot 2023-08-26 at 12 54 43 PM" src="https://github.com/aonez/Keka/assets/6868732/ec91be65-3ce9-450c-8449-0c8a41fc3b78">

^ Sorted by size, it was exactly this I drug in.

**Comment by akrabu** [comment]
I also just extracted that specific file, and checked the CRC32 of it (I embed the CRC32 in the filenames), and it checked out just fine.  So the archive really is fine, it appears!

**Comment by aonez** [comment]
Sure it feels like there is an issue with the verification + password. I'll do some tests tomorrow 🙌🏻

**Comment by akrabu** [comment]
Second attempt verified successfully.  The file hashes ARE different, though.  I'm guessing 7z salts the passwords so that's probably a moot point.

**Comment by aonez** [comment]
I did roughly around 2TBs worth in tests and have not been able to reproduce the issue. Hashes should be different because of the encryption, that is expected. So at this point I'm not sure what the issue was there... I'll recheck the code and leave this issue open in case anyone replicates the issue.

Thanks for all the feedback and test @akrabu 🙇‍♂️

**Comment by gingerbeardman** [comment]
It's a lot of data. Could it be errors from a failing storage device?

**Comment by aonez** [comment]
> Could it be errors from a failing storage device

@gingerbeardman Yeah also thought it may be because of the operation being in an external device so I did my tests using an external device. Maybe you’re right and it was just a temporary read issue, because later tests where valid.

@akrabu how that RAID disk connects to your Mac?

**Comment by akrabu** [comment]
> It's a lot of data. Could it be errors from a failing storage device?

I was concerned about that at first, too.  This is a Samsung T7 Shield 2TB external SSD.  I've had it a little under a year.  It's my main drive for personal photos / videos.  Every file has a CRC32 code in the filename, courtesy of [Rhash](https://github.com/rhash/RHash).  As of last Friday, they also have the SHA256 code AND the last modified date as of summing, stored in extended attributes for each file, courtesy of [Cshatag](https://github.com/rfjakob/cshatag).  I checked with both, and all ~900GB checks out fine.  I don't think it's my SSD.

The RAID-10 array is a SoftRAID volume.  I have six 1TB drives, as a 2TB volume.  The disks themselves are in a 10 bay tower, courtesy of [Sabrent](https://www.amazon.com/gp/product/B09TV1XPDD/).  It's one of several backups (including a cloud backup courtesy of [Restic](https://github.com/restic/restic)), so I'm not concerned I've actually lost anything just yet.

This verification "bug" has also happened to me on non-encrypted archives, I'm fairly certain.  I'll turn on debug mode and just use it like that for awhile.  Perhaps I'll catch it eventually.  It's very sporadic, though.  Happened on both my work and home Macs - but only like 2-3 times since the verification feature was rolled out.  The actual archives always test out fine on a manual check, so it's not ACTUALLY failing.  It's happened on much smaller ~5GB archives, as well.

**Comment by aonez** [comment]
@HiperFall is it possible that _Signal/Linux/Documents/signalbackup-tools-master/signalbackup-tools-master/signal-2021-08-03-03-26-19.backup_ is a symlink?
